addons/sale_stock/models/res_users.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class Users(models.Model):
+    _inherit = ['res.users']
+
+    property_warehouse_id = fields.Many2one('stock.warehouse', string='Default Warehouse', company_dependent=True, check_company=True)
+
+    def _get_warehouse_id(self):
```

Comment:
self.ensure_one

Reply:
When the record is not created yet, we would like to get a default warehouse, so ensure_one would crash on that case, as the recordset is empty. ;)

Reply:
be the res.users exists at this point